examples/cloud-operations/unmanaged-instances-healthcheck/README.md

```diff
@@ -0,0 +1,120 @@
+# TCP healthcheck and restart for unmanaged GCE instances
+
+This example shows how to leverage [Serverless VPC Access](https://cloud.google.com/vpc/docs/configure-serverless-vpc-access) and Cloud Functions to organize a highly performant TCP healtheck for unmanaged GCE instances. Healthchecker Cloud Function uses [goroutines](https://gobyexample.com/goroutines) to achieve parallel healthchecking for multiple instances and handles up to 1 thousand VMs checked in less than a second execution time.
+
+**_NOTE:_** [Managed Instance Groups](https://cloud.google.com/compute/docs/instance-groups/autohealing-instances-in-migs) has autohealing functionality out of the box, current example is more applicable for standalone VMs or VMs in an unmanaged instance group.
+
+The example contains the following components:
+
+- [Cloud Scheduler](https://cloud.google.com/scheduler) to initiate a healthcheck on a schedule.
+- [Serverless VPC Connector](https://cloud.google.com/vpc/docs/configure-serverless-vpc-access) to allow Cloud Functions TCP level access to private GCE instances.
+- **Healthchecker** Cloud Function to perform TCP checks against GCE instances. 
+- **Restarter** PubSub topic to keep track of instances which are to be restarted.
+- **Restarter** Cloud Function to perform GCE instance reset for instances which are failing TCP healthcheck.
+
+
+The resources created in this example are shown in the high level diagram below:
+
+<img src="diagram.png" width="640px">
+
+### Healthchecker configuration
+Healthchecker cloud function has the following configuration options:
+
+- `FILTER` to filter list of GCE instances the health check will be targeted to. For instance `(name = nginx-*) AND (labels.env = dev)`
+- `GRACE_PERIOD` time period to prevent instance check of newly created instanced allowing services to start on the instance. 
+- `MAX_PARALLELISM` - max amount of healthchecks performed in parallel, be aware that every check requires an open TCP connection which is limited.
+- `PUBSUB_TOPIC` topic to publish the message with instance metadata. 
+- `RECHECK_INTERVAL` time period for performing recheck, when a check is failed it will be rechecked before marking as unhealthy. 
+- `TCP_PORT` port used for health checking
+- `TIMEOUT` the timeout time of a TCP probe.
+
+**_NOTE:_** In the current example `healthchecker` is used along with the `restarter` cloud function, but restarter can be replaced with another function like [Pubsub2Mailbox](https://github.com/GoogleCloudPlatform/professional-services/tree/main/tools/pubsub2inbox) for email notifications.
```

Comment:
Pubsub2Mailbox -> Pubsub2Inbox ;)

Reply:
ohhhh, terribly sorry, it was 2am;)) will fix shortly